lxd/provider.go

```diff
@@ -268,6 +277,13 @@ func providerConfigure(d *schema.ResourceData) (interface{}, error) {
 		}
 	}
 
+	// Determine if a custom project was used.
+	// If it wasn't, default to default.
+	project := d.Get("project").(string)
+	if project == "" {
+		project = "default"
```

Comment:
Again this will probably break backwards compatibility. It might be best to only pass "project" if the value is not an empty string.